### DISCUSSION TITLE
duck cli: resume large s3 uploads

### DISCUSSION THREAD

**Discussion by sedlund** [iterate-ch/cyberduck#15316] [Q&A]
It seems the duck cli does not support resuming large file uploads to s3.  Is that correct?

I also found in the gui that stopping a large upload to attempt resume while throttling is enabled hangs the upload.

**Comment by dkocher** [comment]
You should be able to resume uploads using the `--existing resume` command line option.

**Reply by sedlund** [reply]
`--existing resume` does not work for me using https://github.com/iterate-ch/profiles/blob/main/S3%20(Deprecated%20path%20style%20requests).cyberduckprofile with minio.  The gui client resume works.  But I need a CLI tool for Linux that supports resume.  From what I've read --existing resume will continue to upload the files it missed in the previous transfer.

From my understanding of the API and trying to find a CLI client that supports single large file resume. the client has to request the multipart upload - the s3 server issues an upload ID, the client would have to store this.  For the CLI client to resume the upload it would have to tell the server what upload it is resuming.  If the CLI client does support single file resume, where does it store this ID?

If using s3cmd cli - it tells you the job ID to tell it if you want to resume - but s3cmd only uploads one part at a time (no async IO).   So it is very slow.
